scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,118 @@
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v1
+```
+
+The mapping should be updated, so `v1` points to the current release branch reference. Let's say the mapping was:
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `heads/release-4.5`
+
+Then, `v1` is updated to the new release (in this example `release-4.6`):
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `heads/release-4.6`
+
+If in throught 4.6 the bundle version was updated:
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v2
+```
+
+On 4.6 release cut, the final mapping would be:
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `heads/release-4.5`
+`v2` | `heads/release-4.6`
+
+Per-release offline bundles will continue only for 4.5.  New offline bundles will be created per bundle version, and will be replaced by offline bundles per version.  The latest offline bundle will ship all bundle versions currently supported by StackRox, to support customers using the previous workflow.
+
+### Ingesting versioned bundles
+
+Scanner will know the vulnerability version it requires based on hard-coded information added during build time.  A file at `scanner/VULNERABILITY_VERSION` will determine the version on release builds and nightlies.  On non-release builds this will hard-coded to "dev".
+
+The definitions handler in Central can be kept the same.  It already supports arbitrary versions in the GET request from Scanner.  [The check for nightlies](https://github.com/stackrox/stackrox/blob/e332cd19a639f59d9931414a4f4e561981396dad/central/scannerdefinitions/handler/handler.go#L661) and other per-release checks in the ingestion path are unnecessary and will be removed.
+
+The bundle version will be configured in the vulnerability URL by a new variable called `ROX_VULNERABILITY_VERSION` to specify the expected bundle version to fetch.  The previous variable `ROX_VERSION` will be left for backward compatibility.
+
+The vulnreability bundle will include a `manifest.json` file, which will carry the StackRox version and bundle version used to build it.  This file will be used by Scanner to prevent ingestion of unsupported bundle versions.
+
+### Changing schema
+
+When a schema bump is detected (e.g., a ClairCore bump that brings in incompatible schema changes to the vulnerability structs), `scanner/VULNERABILITY_VERSION` should be updated to the new schema version.
+
+Once a StackRox release is out of support, we will remove the entry from the mapping after a grace period, effectively turning off updates for that stream.  The grace period duration will be documented but not specified on this decision record.
+
+## Consequences
+
+- Only necessary vulnerability fetching and parsing workflows will run at every update interval.  This is likely to happen once or twice until better schema handling is implemented in ClairCore.
+- Only necessary bundle formats and versions will exist.
```

Comment:
vs unnecessary? I don't think I fully follow this

Reply:
Fixed it.